File: 백준/Gold/17425. 약수의 합/약수의 합.py
import sys
input = sys.stdin.readline

max = 10 ** 6
fval = [1] * (max + 1)
gval = [0]

def init() :
  for i in range(2, max+1) :
    for j in range(i, max+1, i) :
      fval[j] += i

  for i in range(1, max+1) :
    gval.append(gval[-1] + fval[i])

def query() :
  Q = int(input())
  for _ in range(Q) :
    print(gval[int(input())])

if __name__ == "__main__" :
  init()
  query()

File: 백준/Gold/17425. 약수의 합/약수의 합.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const maxVal int = 1000001

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)
var fval = [maxVal]int{}
var gval = [maxVal]int{}

func initalize() {
	for i := 1; i < maxVal; i++ {
		for j := i; j < maxVal; j += i {
			fval[j] += i
		}
	}
	for i := 1; i < maxVal; i++ {
		gval[i] = gval[i-1] + fval[i]
	}
}

func query() {
	var q, n int
	fmt.Fscanln(reader, &q)
	for i := 0; i < q; i++ {
		fmt.Fscanln(reader, &n)
		fmt.Fprintln(writer, gval[n])
	}
}
func main() {
	defer writer.Flush()
	initalize()
	query()
}
